test/lit.cfg

```diff
@@ -930,39 +930,68 @@ elif run_os in ['linux-gnu', 'linux-gnueabihf', 'freebsd', 'windows-cygnus', 'wi
         "clang++ -target %s %s" %
         (config.variant_triple, clang_mcp_opt))
     config.target_ld = "ld -L%r" % (make_path(test_resource_dir, config.target_sdk_name))
-elif run_os == 'linux-androideabi':
+elif run_os == 'linux-androideabi' or run_os == 'linux-android':
+    def get_architecture_value(**kwargs):
+        result = kwargs[run_cpu]
+        if result is None:
+          if run_cpu == "armv7s" or run_cpu == "armv7k":
+            result = kwargs["armv7"]
+          elif run_cpu == "arm64":
+            result = kwards["aarch64"]
+        return result
+
+    ndk_platform_tuple = get_architecture_value(armv7="armeabi-v7a",
+                                                aarch64="arm64-v8a")
+    ndk_platform_triple = get_architecture_value(armv7="arm-linux-androideabi",
+                                                 aarch64="aarch64-linux-android")
+    toolchain_directory_name = "{}-{}".format(ndk_platform_triple, config.android_ndk_gcc_version)
+    tools_directory = make_path(config.android_ndk_path, "toolchains",
+        toolchain_directory_name, "prebuilt", "linux-x86_64",
```

Comment:
`linux-x86_64` should really be based on the build (e.g. building on Windows, Darwin, or Linux)